rest/method_get.go

```diff
@@ -43,7 +43,7 @@ func listGet(ctx context.Context, r *http.Request, route *RouteMatch) (status in
 		list.Offset = win.Offset
 	}
 	for _, item := range list.Items {
-		item.Payload, err = q.Projection.Eval(ctx, item.Payload, rsc.Validator(), getReferenceResolver(rsc))
+		item.Payload, err = q.Projection.Eval(ctx, item.Payload, restResource{rsc})
```

Comment:
After some reflectionI think `query.Resource` is an overall better idea, and easier to understand.

It _might_ also be that with some further refactoring of resource.Resource, it _could_ be made to implement `query.Resource`, which would simply usage. The `resource.Item` type would plausibly have to be dropped (or moved) to achieve this.